cmd/infra/powervs/destroy.go

```diff
@@ -533,9 +597,18 @@ func destroyVpcSubnet(options *DestroyInfraOptions, infra *Infra, resourceGroupI
 	return pagingHelper(f)
 }
 
-// deleteVpcSubnet deletes the subnet id passed
-func deleteVpcSubnet(id string, v1 *vpcv1.VpcV1, infraID string) (err error) {
-	log(infraID).Info("Deleting VPC subnet", "subnetId", id)
+// deleteVpcSubnet deletes the subnet id passed and LB attached to it
+func deleteVpcSubnet(id string, v1 *vpcv1.VpcV1, options *DestroyInfraOptions) (err error) {
+	// deleting the load balancer before proceeding to subnet deletion, since LB is an attached resource to the subnet
+	e := destroyVpcLB(options, id, v1)
+	if e != nil {
+		log(options.InfraID).Error(e, "error destroying VPC Load Balancer")
+		err = fmt.Errorf("error destroying VPC Load Balancer %w", e)
```

Comment:
to be consistent with the rest of the code base, can we just return directly `return fmt.Errorf("error destroying VPC Load Balancer %w", e)`

Reply:
returned the error directly.